aliri_tokens/src/sources/oauth2.rs

```diff
@@ -149,13 +166,19 @@ async fn request_token<R: CredentialsSource, C: Clock>(
     token_url: reqwest::Url,
     credentials: &mut R,
     lifetime_config: &TokenLifetimeConfig<C>,
+    content_type: &ContentType
 ) -> Result<TokenWithLifetime, TokenRequestError> {
     tracing::trace!("requesting token from authority");
 
-    let resp = client
-        .post(token_url)
-        .json(&credentials)
-        .send()
+    let mut req = client
+        .post(token_url);
+    if *content_type == ContentType::Form {
+        req = req.form(&credentials);
+    }
+    else {
+        req = req.json(&credentials);
+    }
```

Comment:
This should instead be a `match` expression to ensure that any future content type additions would be taken care of appropriately.

We may also be able to deal with this with a generic parameter `BodyType` that defaults to `Json` and instead can be specified as `Form`. This would give the same result as what you've done here, but without requiring a branch for every call.